File: CS/ViewerCustomCommand/Form1.cs
using DevExpress.XtraPdfViewer;
using System;
using System.Windows.Forms;

namespace ViewerCustomCommand {
    public partial class Form1 : Form {

        void ReplacePdfViewerCommandFactoryService(PdfViewer control) {
            IPdfViewerCommandFactoryService service = control.GetService<IPdfViewerCommandFactoryService>();
            if (service == null)
                return;
            control.RemoveService(typeof(IPdfViewerCommandFactoryService));
            control.AddService(typeof(IPdfViewerCommandFactoryService), new CustomPdfViewerCommandFactoryService(control,service));
        }

        public Form1() {
            InitializeComponent();
            pdfViewer1.LoadDocument("..\\..\\Demo.pdf");
            ReplacePdfViewerCommandFactoryService(pdfViewer1);
        }     
    }
}

File: VB/ViewerCustomCommand/Form1.vb
Imports DevExpress.XtraPdfViewer
Imports System.Windows.Forms

Namespace ViewerCustomCommand

    Public Partial Class Form1
        Inherits Form

        Private Sub ReplacePdfViewerCommandFactoryService(ByVal control As PdfViewer)
            Dim service As IPdfViewerCommandFactoryService = control.GetService(Of IPdfViewerCommandFactoryService)()
            If service Is Nothing Then Return
            control.RemoveService(GetType(IPdfViewerCommandFactoryService))
            control.AddService(GetType(IPdfViewerCommandFactoryService), New CustomPdfViewerCommandFactoryService(control, service))
        End Sub

        Public Sub New()
            InitializeComponent()
            pdfViewer1.LoadDocument("..\..\Demo.pdf")
            ReplacePdfViewerCommandFactoryService(pdfViewer1)
        End Sub
    End Class
End Namespace
